src/azure-cli/azure/cli/command_modules/redis/custom.py

```diff
@@ -35,6 +35,15 @@ def cli_redis_update(cmd, instance, sku=None, vm_size=None):
         instance.sku.family = vm_size[0]
         instance.sku.capacity = vm_size[1:]
 
+    # avoid setting memory configs for basic sku
+    if instance.sku.name == 'Basic':
+        if 'maxmemory-reserved' in instance.redis_configuration:
+            instance.redis_configuration.pop('maxmemory-reserved')
+        if 'maxfragmentationmemory-reserved' in instance.redis_configuration:
+            instance.redis_configuration.pop('maxfragmentationmemory-reserved')
+        if 'maxmemory-delta' in instance.redis_configuration:
+            instance.redis_configuration.pop('maxmemory-delta')
+
```

Comment:
Why those three property values of `maxmemory-reserved`, `maxfragmentationmemory-reserved`, `maxmemory-delta` should be returned?
I mean those property values seem like shouldn't belong to the instance created with Basic SKU, should be trimmed.

Reply:
These property values are read-only for Basic SKU cache, so they need to be returned but they cannot be updated.

Reply:
These three values are allowed for Standard and Premium SKU?

Reply:
Yes. they are available for Standard and Premium SKUs. Documentation link: <https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-configure#memory-policies>

Reply:
I know. But I insist that these property values should be invisble for user and service-side should handle those exception Instead of cover that in the frontend side.

Imagine that, if user directly call API to manipulate Redis and he/she shoule get the same warning from service side because he/she saw those properties before and than he/shee finally aware of thoses properties are read-only for current SKU. What if those properties got trimmed at the first place? What if service-side can deal with those unnecessary properties for Basic SKU?

Those values are not present in azure-sdk-for-python too.  azure-cli can cover those details for service-side easily but I think too much.

Reply:
Service side cannot hide these properties on GET call, as these properties needs to be displayed to user and are currently displayed as uneditable fields in azure portal.
If user directly call API with these properties, service side will throw appropriate error. We cannot swallow this error, as user might think that they are able to set these properties.

Reply:
So there is no better way to walk it around. It would be better to do it in azure-cli